src/framework/script/script-registry.js

```diff
@@ -110,13 +135,15 @@ class ScriptRegistry extends EventHandler {
 
                     if (scriptInstance)
                         scriptInstances.push(scriptInstance);
+
+                    // initialize attributes
+                    for (let i = 0, len = component.scripts.length; i < len; i++) {
+                        const script = component.scripts[i];
+                        component.initializeAttributes(script);
+                    }
```

Comment:
Since this isn't performance critical code, I'd just do:

```
                for (const script of component.scripts) {
                    component.initializeAttributes(script);
                }
```